apps/enc.c

```diff
@@ -418,15 +426,31 @@ int enc_main(int argc, char **argv)
                     BIO_printf(bio_err, "bad magic number\n");
                     goto end;
```

Comment:
I think we should extend the magic number handling above. We should introduce a new magic number that if provided will specify all the relevant parameters, e.g. password algorithm (PBKDF2), cipher, digest, number of iterations, salt etc. We may need other params in the future. Old versions of OpenSSL will fail with a "bad magic number" error if they encounter one of the new format files...but that's ok, they're going to fail anyway with this change.